### ISSUE TITLE
Full-text search

### ISSUE THREAD

**Issue by shaobozi** [jrnl-org/jrnl#463]

**Comment by maebert** [comment]
Hey Shabozi, can you maybe give an example of what you mean?

**Comment by shaobozi** [comment]
like the command >: jrnl -s "blah" ; will list all entries with the word "blah" .

**Comment by maebert** [comment]
Ah, now I understand.This'll come in the next big release, see also #313.

**Comment by radomirbosak** [comment]
I rebased my earlier PR which solves this: #470 

@shaobozi Note that `-s` is already used for turning on short output (`--short`).

@maebert , do you plan to postpone any search-related features until 2.2 ? What is the current schedule for adding new features?

**Comment by craftyguy** [comment]
Is this feature still planned? Being able to search for any text in a journal would be super helpful, and it would save me from having to @tag every word :P

**Comment by jyoo** [comment]
> Ah, now I understand.This'll come in the next big release, see also #313.

Is this feature still slated for release? Been almost 2 years.

**Comment by wren** [comment]
Yup, this is a good idea for a feature. We'll (re)schedule it for a milestone once we get the rest of the project organized. 

PRs welcome!

**Comment by dradux** [comment]
This has been a blocker for me to use the app in any serious way for over 2 years. Perhaps I am not the target user but search seems to be the cornerstone to journals/notes (assuming you intend to use them in any meaningful way in the future). You will likely need to find something you wrote in the past and relying on adding a tag for everything you _may_ want search on does not seem feasible. Do typical users fall back to grep for this or build their own lucene index?

**Comment by wren** [comment]
@dradux You'll be happy to know that Search is one of our top priorities and now scheduled for an upcoming milestone.

I can't speak for most users (although I'd love to hear other people's approaches), but my personal approach is to open the whole journal in neovim when I want to do any searching of this kind. I do this because I keep most of my journals encrypted, so grep isn't super useful. I use ripgrep for the unencrypted journals sometimes. Either way, it'd be useful to have something built in for users. 

As for building an index, the plain text format, and the encryption feature come together to make building and keeping any sort of metadata (including an index) very tricky in jrnl. I consider both of those features integral to jrnl, so metadata has been a difficult topic to approach, and search will likely depend on it in some way or another (unless we want to search the entire journal every time, which can lead to some performance problems in larger journals).

All of that is to say that we have a milestone to consider the exact format we want to use in jrnl going forward (which this will include considerations for metadata while maintaining the plain text and encryption features). This needs to happen before we implement a feature like search, but both are definitely in the pipeline.

**Comment by empireshades** [comment]
any update on this one?

**Comment by pspeter** [comment]
Nobody has submitted a PR yet. It is scheduled for a future release already, but there's some other issues/features that were prioritized. You are welcome to submit a PR if you want to have this feature in sooner.

**Comment by empireshades** [comment]
I just submitted [](https://github.com/jrnl-org/jrnl/pull/740). 🤞
